### PR TITLE
Refactor gamepads layotus and add some more

### PR DESCRIPTION
**Pull request by arielj**
This PR is an updated to the layouts handling for different gamepads. Changes are:

- Reorganized the layouts in a gamepad_layouts folder grouping layouts by brand
- Added Genius MaxFire, a N64-USB clone, ps3 clone through usb dongle
- Prevent exceptions when checking buttons to break the gamepad loop

I was able to test the Genius layout and will ask Leopard to test the ps3 one.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/helpers/gamepad.ts [hunk 1]
```diff
@@ -217,191 +226,35 @@ export const initGamepad = () => {
...
+        } else if (controller.id.match(/0583.*a009/i)) {
+          checkGenius1(buttons, axes, index, checkAction)
+        } else {
+          // if not specific, fallback to the xbox layout, seems
+          // to be the most common for now and if not exact it seems
+          // to cover at least the left stick and the main 2 buttons
+          checkXbox(buttons, axes, index, checkAction)
+        }
+      } catch (error) {
+        console.log(`Gamepad error: ${error}`)
```
Comment (Nocccer) [inline src/helpers/gamepad.ts:251]: Please use our log functions. (logger.ts).

You could also write a wrapper around it.
e.g.:

```typescript
const gamepadLog = (type: 'info' | 'warning' | 'error', message: string) => {
  const extMessage = `Gamepad: ${message}`
  switch(type) {
    case 'info':
      logInfo(extMessage)
      break
   case 'warning':
      logWarning(extMessage)
      break
   case 'error':
     logError(extMessage)
     break
   default:
     break
  }
}
```

File: src/helpers/gamepad.ts [hunk 2]
```diff
@@ -414,8 +267,10 @@ export const initGamepad = () => {
         console.log(`button ${button} pressed ${buttons[button].value}`)
     }
     for (const axis in axes) {
-      if (axes[axis] < -0.5) console.log(`axis ${axis} activated negative`)
-      if (axes[axis] > 0.5) console.log(`axis ${axis} activated positive`)
+      if (axes[axis] < -0.1 && axes[axis] >= -1)
+        console.log(`axis ${axis} activated negative`)
```
Comment (Nocccer) [inline src/helpers/gamepad.ts:271]: use our logger or a wrapper i mentioned above

File: src/helpers/gamepad.ts [hunk 3]
```diff
@@ -414,8 +267,10 @@ export const initGamepad = () => {
...
+      if (axes[axis] > 0.1 && axes[axis] <= 1)
+        console.log(`axis ${axis} activated positive`)
```
Comment (Nocccer) [inline src/helpers/gamepad.ts:273]: use our logger or a wrapper i mentioned above

**Review by Nocccer — CHANGES_REQUESTED** [review]
Left some suggestions there for better logging.

**Review by flavioislima — APPROVED** [review]
Looks good!

File: src/helpers/gamepad.ts (hunk 1 above)
Comment (flavioislima) [reply src/helpers/gamepad.ts:251]: This is actually on the frontend, will they work there as well? I think they are on the electron folder so cannot be imported here.

Comment (Nocccer) [reply src/helpers/gamepad.ts:251]: True. So if we want this message on the terminal we should use ipcRedender('LogInfo' ...), else we keep it like that.

**Review by Nocccer — APPROVED** [review]

**Comment by arielj** [comment]
thanks! I'll update the logger calls after that's merged on the main branch :+1: (I noticed it got merged but something went wrong there during the merge)

**Review by flavioislima — APPROVED** [review]
LGTM! 👍🏽

**Pull request merged**